api/app/events/api.py

```diff
@@ -306,22 +306,25 @@ def put(self):
             LOGGER.error("Event with KEY: {} already exists".format(key))
             return EVENT_KEY_IN_USE
 
-        event_org = event_repository.get_by_id_with_organisation(event.id)
-        return event_details(event_org), 200
+        event = event_repository.get_by_id(event.id)
+        return event, 200
 
 
 class EventsAPI(restful.Resource):
 
     @auth_required
     def get(self):
         user_id = g.current_user["id"]
+        language = request.args['language']
 
         events = event_repository.get_upcoming_for_organisation(g.organisation.id)
         returnEvents = []
 
         for event in events:
-            status = None if user_id == 0 else event_status.get_event_status(event.Event.id, user_id)
-            returnEvents.append(event_info(user_id, event, status))
+            if not event.has_specific_translation(language):
+                return EVENT_WITH_TRANSLATION_NOT_FOUND
```

Comment:
@avishkar58 Wasn't sure how to error handle this part specifically since if even just one of the upcoming events doesn't have the specified language, the response will return not found?
So should I keep it like this? Default to English (i.e. will all events going forward always have an English translation)? Skip the event entirely (this doesn't feel right though)? Or return the event with the name and description blank (and log errors so we're alerted)?

Reply:
Hey @amritpurshotam - good question... I think it's fair to assume for now that all events will have an English translation, so how about we default to English and log an error?

Reply:
@avishkar58 Cool will do that then 👍